contracts/chain_test.go

```diff
@@ -132,8 +132,31 @@ func (s *storeMock) Contract(_ context.Context, contractID types.FileContractID)
 	return Contract{}, ErrNotFound
 }
 
-func (s *storeMock) Contracts(ctx context.Context, opts ...ContractQueryOpt) ([]Contract, error) {
-	return slices.Clone(s.contracts), nil
+func (s *storeMock) Contracts(ctx context.Context, queryOpts ...ContractQueryOpt) ([]Contract, error) {
+	cloned := slices.Clone(s.contracts)
```

Comment:
No benefit to cloning the slice here since it's going to be filtered below.


```suggestion
    var opts ContractQueryOpts
	for _, opt := range queryOpts {
		opt(&opts)
	}

	filtered := make([]Contract, 0, len(s.contracts))
	for _, c := range s.contracts {
		if opts.Revisable != nil {
			isRevisable := (c.State == ContractStatePending || c.State == ContractStateActive) && c.RenewedTo == (types.FileContractID{})
			if isRevisable != *opts.Revisable {
				continue
			}
		}
		if opts.Good != nil {
			if c.Good != *opts.Good {
				continue
			}
		}
		filtered = append(filtered, c)
	}
```

Reply:
removing in another PR 👍